src/widgets/settingspages/GeneralPage.cpp

```diff
@@ -877,6 +877,18 @@ void GeneralPage::initLayout(GeneralPageView &layout)
     helixTimegateModerators->setMinimumWidth(
         helixTimegateModerators->minimumSizeHint().width());
 
+    auto *helixTimegateViewerList =
+        layout.addDropdown<std::underlying_type<HelixTimegateOverride>::type>(
+            "Helix timegate viwer list behaviour",
+            // AlwaysUseIRC value is used to always use tmi endpoint
+            {"Timegate", "Always use tmi", "Always use Helix"},
+            s.helixTimegateViewerList,
+            helixTimegateGetValue,  //
+            helixTimegateSetValue,  //
+            false);
+    helixTimegateViewerList->setMinimumWidth(
+        helixTimegateViewerList->minimumSizeHint().width());
```

Comment:
#### Don't feel any rush to action on this, I'd like some thoughts on my comment.

I would like this to use `AlwaysUseIRC` by default, because as of right now it seems the TMI viewer list will not be deprecated along with IRC commands.
`(It was announced separately and I do believe they said on stream the date would be announced "in the future", so it's unlikely)`

My main reason for this approach is due to the fact that new nightly users are generated daily, and this would allow them the most seamless continuation, and not require them to update, as this is the final IRC deprecation change.

If the deprecation does not happen in Feb, the list continues to work exactly as expected.

If the deprecation DOES happen, the following will happen:
- The viewer list will not work for non-mods, and there's nothing they can do about this.
- The viewer list will not work for mods or the broadcaster, but they can ratify this by swapping the setting to `Always Use Helix`


Comparing both scenarios, I believe less nightly users will be impacted taking the `AlwaysUseIRC` approach. (every single mod/broadcaster versus every single chatter)


<!---
Do we think the correct approach for this would be to skip the time-gate and default to `AlwaysUseIRC`?

Due to the fact that the chatters deprecation is very likely not apart of the IRC Commands deprecation, I believe it would be better to force use the TMI list.
`(It was announced separately and I do believe they said on stream the date would be announced "in the future")` 

My main reason for this approach 
If the deprecation does not happen in Feb, the list continues to work exactly as expected.

If the deprecation DOES happen, the following will happen:
- The viewer list will not work for non-mods, and there's nothing they can do about this.
- The viewer list will not work for mods or the broadcaster, but they can ratify this by swapping the setting to `Always Use Helix`

Comparing both scenarios, I believe less people will be impacted taking the `AlwaysUseIRC` approach. (every single mod/broadcaster versus every single chatter)

This would also allow for nightly users from now until Feb to just not update, 
--->

Reply:
That makes sense. Should I just takeout the timed portion all together and just allow switching between helix and tmi with the tmi being the default?

Reply:
You can probably take the time-gate out, but I'd wait for somebody else's opinion as they might disagree with me.

Reply:
I talked with pajlada about this. We basically came to the conclusion that using a timebomb for this is counterproductive and doesn't solve anything. I came up with the idea of adding an endpoint on the chatterino api and having a remote kill switch, when we know the TMI api is dead, we can just disable it on Chatterinos of willing users.

Reply:
I think you should just remove the timeout section of this PR, yes.

If we want to look at some other solution to make it better for non-broadcaster users, it should be done in another PR imo.

Reply:
@pajlada So should there be a switch between the tmi and the helix requests? We can just keep the old function as is like I have it now.

Reply:
I don't think we should keep any of the old "TMI Chatters" functionality. Remove the time gating that's been implemented in this PR and purely run requests of Helix and fail if the user does not have permission